src/fu-engine.c

```diff
@@ -643,32 +694,34 @@ fu_engine_verify (FuEngine *self, const gchar *device_id, GError **error)
 		return FALSE;
 
 	/* find component in metadata */
-	store = fu_engine_load_verify_store (error);
-	if (store == NULL)
-		return FALSE;
-	app = fu_engine_store_get_app_by_guids (store, device);
-	if (app == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_NOT_FOUND,
-				     "No metadata");
-		return FALSE;
+	version = fu_device_get_version (device);
+	localstatedir = fu_common_get_path (FU_PATH_KIND_LOCALSTATEDIR_PKG);
+	fn = g_strdup_printf ("%s/verify/%s.xmlb", localstatedir, device_id);
```

Comment:
`xmlb` files seem like they should be going into a cache directory not a local state directory.  Am I mistaken?

Reply:
Not mistaken, perhaps misled :) Most of the time an XMLB file is a per-user cache -- here I'm (mis-?)using it as a compressed metadata store. Perhaps we should be saving the plain exported XML to the /verify/ directory rather than the compressed blob, although then we have to parse it again when verifying. You can always export the xmlb as an XML file manually using xb-tool, although it's perhaps a little obscure.

Reply:
Assuming you saved the XML file to /verify/ shouldn't it be quick though if there is already a cached file in the cache directory?  It seems like the correct model to me.